libs/base/utilities/src/misc/events-repeater.ts

```diff
@@ -0,0 +1,134 @@
+const eventsDataIdentifier = 'EVENTS_DATA';
+export const EVENTS_DATA = Symbol.for(eventsDataIdentifier);
+export const repeatableAttribute = 'repeatable';
+
+export type ElementWithEventsData = HTMLElement & {
+  [EVENTS_DATA]?: EventsData;
+};
+
+interface EventsData {
+  listener: (event: Event) => void;
+  events: Event[];
+}
+
+export function treewalk(
+  selector: string,
+  rootNode = document.body,
+  includeRoot = true
+): HTMLElement[] {
+  const nodes: Element[] = [rootNode];
+  const elements: HTMLElement[] = [];
+
+  for (const node of nodes) {
+    if (!node) continue;
+
+    if (node.nodeType !== Node.ELEMENT_NODE) {
+      continue;
+    }
+
+    if (node.children.length) {
+      nodes.push(...node.children);
+    }
+
+    if (node.shadowRoot?.children.length) {
+      nodes.push(...node.shadowRoot.children);
+    }
+
+    if (!includeRoot && node.matches(rootNode.tagName.toLowerCase())) {
+      continue;
+    }
+
+    if (node.matches(selector)) {
+      elements.push(node as HTMLElement);
+    }
+  }
+
+  return elements;
+}
+
+function eventsAction(
+  element: ElementWithEventsData,
+  shouldRemove?: true
+): void {
+  const events = element.getAttribute(repeatableAttribute) as string;
+
+  for (const eventType of events.split(',')) {
+    if (shouldRemove) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      element.removeEventListener(eventType, element[EVENTS_DATA]!.listener);
```

Comment:
We should probably trim the `eventType`, same a few lines below. And add a test for it.